src/Artsy/Relay/renderWithLoadProgress.tsx

```diff
@@ -17,56 +17,45 @@ const SpinnerContainer = styled.figure`
 
 export const LoadingClassName = "relay-loading"
 
+const handleError = error => {
+  // In tests we want errors to clearly bubble up.
+  if (typeof jest !== "undefined") {
+    throw error
+  }
+
+  if (error.message) {
+    console.error(error.message)
+  }
+
+  const networkError = error as any
+  if (networkError.response && networkError.response._bodyInit) {
+    let data = networkError.response._bodyInit
+    if (data) {
+      try {
+        data = JSON.parse(data)
+        console.error(`Metaphysics Error data:`, data)
+        // tslint:disable-next-line:no-empty
+      } catch (e) {}
+    }
+  }
+}
+
 export function renderWithLoadProgress<P>(
   Container: RelayContainer<P>,
-  initialProps: object = {}
+  initialProps: object = {},
+  Wrapper: any = SpinnerContainer,
```

Comment:
`Wrapper` feels a little nonspecific to me, since it's only used for wrapping the loading spinner. How about `SpinnerWrapper`? Or why not even just let consumers specify the entire thing by making this `LoadingComponent: React.ComponentType<{}> = ...`

and then instead of returning

        <Wrapper className={LoadingClassName}>
          <Spinner />
        </Wrapper>

you just return

        <LoadingComponent />

Then you could have

    const DefaultSpinner = () => (
        <SpinnerContainer className={LoadingClassName}>
          <Spinner />
        </SpinnerContainer>
    )

and 

`LoadingComponent: React.ComponentType<{}> = DefaultSpinner`

Reply:
This is good feedback and a solid approach, but I ended up going with the easier solution of just accepting a props object for the container. Over time we'll see if that's good enough or if we end up wanting more control.